mirrord/layer/src/detour.rs

```diff
@@ -353,3 +353,24 @@ impl<T> OptionExt for Option<T> {
         }
     }
 }
+
+pub(crate) trait OnceLockExt<T> {
+    fn get_or_detour_init<F>(&self, f: F) -> Detour<&T>
+    where
+        F: FnOnce() -> Detour<T>;
+}
+
+impl<T> OnceLockExt<T> for OnceLock<T> {
+    fn get_or_detour_init<F>(&self, f: F) -> Detour<&T>
+    where
+        F: FnOnce() -> Detour<T>,
+    {
+        if let Some(value) = self.get() {
```

Comment:
Chains?

Or at the very least I would suggest removing `return` and using an `else` clause.

Reply:
Actually this is the only part left from the implementation of `get_or_try_init` but sure Ill change it to an else to prevent complexity with ?